packages/core/src/render3/features/ng_onchanges_feature.ts

```
@@ -83,12 +84,12 @@ function rememberChangeHistoryAndInvokeOnChangesHook(this: OnChanges) {
 
 function ngOnChangesSetInput<T>(
     this: DirectiveDef<T>, instance: T, value: any, publicName: string, privateName: string): void {
+  const declaredName = (this.declaredInputs as {[key: string]: string})[publicName];
+  ngDevMode && assertString(declaredName, 'Name of input in ngOnChanges has to be a string');
```

I added this assertion, because without it `ngOnChanges` can fail silently by creating a `SimpleChanges` object that looks like `{undefined: {...}}`. This threw me off initially and took me a while to figure out.